[vannykin] iP

vannykin opened the pull request:
LumiereMax frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,


1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

vannykin committed:
Level-3

vannykin committed:
Level-4

vannykin committed:
Level-5

vannykin committed:
Level-6

vannykin committed:
Level-7

vannykin committed:
Level-8

ZHANGTIANYAO1 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,28 @@
+        this.deadline = LocalDate.parse(byWhen);
+    }
+
+    public String getbyWhen() {
```
Should this be getByWhen?

ZHANGTIANYAO1 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,28 @@
+    }
+
+    @Override
+    public String stringify() {
```
What does `stringify `mean? I know there is a stringify fun in JavaScript for JSON things, does it also defined in java?

angsongyee replied:
I think the method they were looking for might be `toString()`? Based on how this method is used I think that would be a more appropriate name. Also, another benefit of overriding `toString()` instead of writing an entirely new method is because then Java can automatically call it whenever the object needs to be converted to a String (i.e. string concatenation or passing objects as arguments to `System.out.println()`).
```suggestion
    public String toString() {
```

ZHANGTIANYAO1 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,28 @@
+    private String byWhen;
+    private LocalDate deadline;
+
+    public Deadline(String item, boolean marked, String byWhen) {
```
Maybe use `isMarked`?

ZHANGTIANYAO1 commented on src/main/java/Lumiere.java:
```diff
@@ -0,0 +1,163 @@
+import java.util.*;
```
Might be better to avoid using * to import everything in the package.

ZHANGTIANYAO1 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,29 @@
+public class Task {
+    private boolean marked;
```
Should this be `isMarked`?

ZHANGTIANYAO1 commented on src/main/java/lumiere.txt:
```diff
@@ -0,0 +1,2 @@
+T | false | finish up part 4
+D | false | decide on rs | 2024-02-29
```
May not need to upload this file?

ZHANGTIANYAO1 reviewed (commented):
Overall LGTM 👍 
Just some small nits to fix.

angsongyee commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,28 @@
+    @Override
+    public String stringify() {
+        String m = "";
+        if (super.isMarked())
+            m = "[X]";
+        else
+            m = "[ ]";
+        return "[D]" + m + " " + super.stringify() + " (by: "
+                + this.deadline.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
+    }
```
It's not necessary to use the keyword `super` here, and I think it makes the code harder to read. On the other hand, `super.stringify()` is definitely an appropriate usecase for it so good job for that👍. Basically, you should only use `super` when you need the superclass's implementation of the method instead of the one in your current class.
```suggestion
        if (isMarked())
            m = "[X]";
        else
            m = "[ ]";
        return "[D]" + m + " " + super.stringify() + " (by: "
                + this.deadline.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
    }
```

angsongyee commented on src/main/java/Lumiere.java:
```diff
@@ -0,0 +1,163 @@
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+
+        List<Task> list = loadTasksFromFile();
```
Maybe a more descriptive name could be used here? 
```suggestion
        List<Task> tasks = loadTasksFromFile();
```

angsongyee commented on src/main/java/Lumiere.java:
```diff
@@ -0,0 +1,163 @@
+            if (command.equals("list")) {
+                System.out.println("Here are the tasks in your list:");
+                for (int i = 0; i < list.size(); i++) {
+                    System.out.println(Integer.toString(i + 1) + "." + list.get(i).stringify());
```
As mentioned above, if you override `toString()` instead of using `stringify()` you should be able to print your tasks like this.
```suggestion
                    System.out.println(Integer.toString(i + 1) + "." + list.get(i));
```

angsongyee commented on src/main/java/Todo.class:
In addition to `lumiere.txt`, `.class` files do not need to be committed to the repo. You should try to modify your `.gitignore` to exclude these types of files from your commits.

angsongyee reviewed (commented):
Code looks good, there's only a few minor style issues. Keep up the good work 👍.

vannykin committed:
A-MoreOOP

vannykin committed:
A-Packages

vannykin committed:
Merge remote-tracking branch 'origin/add-gradle-support'

vannykin committed:
A-Gradle + A-JUnit

vannykin committed:
A-Jar

vannykin committed:
Level-10

vannykin committed:
A-JavaDoc

vannykin committed:
Merge branch 'branch-A-JavaDoc'

vannykin committed:
A-CodingStandard

vannykin committed:
Actual-Level-10

vannykin committed:
Merge branch 'branch-Level-10'

vannykin committed:
Level-10-Actual

vannykin committed:
Level-10-Final

vannykin committed:
This commit is non-trivial.

This commit is to include / update the new JAR file I created, as well
as for me to learn how to write full commit message.

vannykin committed:
A-Assertions

vannykin committed:
A-CodeQuality

vannykin committed:
Merge pull request #2 from vannykin/branch-A-Assertions

PR-A-Assertions

vannykin committed:
Merge remote-tracking branch 'origin/master' into branch-A-CodeQuality

vannykin committed:
Merge pull request #3 from vannykin/branch-A-CodeQuality

PR-A-CodeQuality

vannykin committed:
BCD-Extension

vannykin committed:
A-UserGuide

vannykin committed:
A-Release

vannykin committed:
Update Ui.png name

vannykin committed:
Fix bug - final submission
